### DISCUSSION TITLE
Failed to start Z2M after updating Tasmota > 13.2.0

### DISCUSSION THREAD

**Discussion by oywino** [arendst/Tasmota#22129] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I have had Tasmota running on a Sonoff ZigBee Bridge for a long time. I have Z2M running in a docker container connected to the Zonoff bridge and working well. I run Mosquitto MQTT in a docker container also.
Everything has been working well with a number of ZigBee devices contected, up and until version 13.2.0 of Tasmota. 
All Tasmota later than 13.2.0 breaks the connection from Z2M to the ZB Bridge. I have tried several different Tasmota versions.
Every time when I revert back to Tasmota 13.2.0 it all works again.

The error log from my Z2M shows the same independent of the version of Z2M (currently v1.40.1):
```
z2m-zigbee2mqtt-1  | [2024-09-11 16:18:41] error:       z2m: Error while starting zigbee-herdsman
z2m-zigbee2mqtt-1  | [2024-09-11 16:18:41] error:       z2m: Failed to start zigbee
z2m-zigbee2mqtt-1  | [2024-09-11 16:18:41] error:       z2m: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
z2m-zigbee2mqtt-1  | [2024-09-11 16:18:41] error:       z2m: Exiting...
z2m-zigbee2mqtt-1  | [2024-09-11 16:18:41] error:       z2m: Error: Failure to connect
z2m-zigbee2mqtt-1  |     at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:364:19)
z2m-zigbee2mqtt-1  |     at processTicksAndRejections (node:internal/process/task_queues:95:5)
z2m-zigbee2mqtt-1  |     at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:183:13)
z2m-zigbee2mqtt-1  |     at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:138:29)
z2m-zigbee2mqtt-1  |     at Zigbee.start (/app/lib/zigbee.ts:64:27)
z2m-zigbee2mqtt-1  |     at Controller.start (/app/lib/controller.ts:140:27)
z2m-zigbee2mqtt-1  |     at start (/app/index.js:154:5)
z2m-zigbee2mqtt-1 exited with code 1
z2m-zigbee2mqtt-1  | Starting Zigbee2MQTT without watchdog.
z2m-zigbee2mqtt-1  | [2024-09-11 16:18:51] error:       zh:ezsp:ezsp: Connection attempt 1 error: Error: connect ECONNREFUSED 192.168.1.115:8888
z2m-zigbee2mqtt-1  | [2024-09-11 16:18:56] error:       zh:ezsp:ezsp: Connection attempt 2 error: Error: connect ECONNREFUSED 192.168.1.115:8888
z2m-zigbee2mqtt-1  | [2024-09-11 16:19:06] error:       zh:ezsp:ezsp: Connection attempt 3 error: Error: connect ECONNREFUSED 192.168.1.115:8888
z2m-zigbee2mqtt-1  | [2024-09-11 16:19:21] error:       zh:ezsp:ezsp: Connection attempt 4 error: Error: connect ECONNREFUSED 192.168.1.115:8888
z2m-zigbee2mqtt-1  | [2024-09-11 16:19:21] error:       z2m: Error while starting zigbee-herdsman
z2m-zigbee2mqtt-1  | [2024-09-11 16:19:21] error:       z2m: Failed to start zigbee
z2m-zigbee2mqtt-1  | [2024-09-11 16:19:21] error:       z2m: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
z2m-zigbee2mqtt-1  | [2024-09-11 16:19:21] error:       z2m: Exiting...
```

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): Sonoff ZHA Bridge
- [ ] Tasmota binary firmware version number used: 14.2.0
  - [X] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: TasmoAdmin
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
```
16:46:47.176 MQT: stat/sonoff_zb_bridge/RESULT = {"NAME":"Sonoff ZHABridge","GPIO":[320,5088,0,5120,323,322,0,0,0,0,0,0,32,0],"FLAG":0,"BASE":18}
16:46:47.389 MQT: stat/sonoff_zb_bridge/RESULT = {"Module":{"0":"Sonoff ZHABridge"}}
16:46:47.645 MQT: stat/sonoff_zb_bridge/RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"5088":"TCP Tx"},"GPIO2":{"0":"None"},"GPIO3":{"5120":"TCP Rx"},"GPIO4":{"323":"Led_i4"},"GPIO5":{"322":"Led_i3"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"32":"Button1"},"GPIO17":{"0":"None"}}
16:47:08.208 MQT: tele/sonoff_zb_bridge/STATE = {"Time":"2024-09-11T16:47:08","Uptime":"0T00:30:09","UptimeSec":1809,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Telia-1D882E","BSSId":"A2:B5:3C:1D:68:2E","Channel":4,"Mode":"11n","RSSI":100,"Signal":-29,"LinkCount":1,"Downtime":"0T00:00:04"}}
```
```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
stat/sonoff_zb_bridge/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":37,"Free":474,"Rules":"ON System#Boot do TCPStart 8888 endon"}}

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
16:50:01.793 MQT: stat/sonoff_zb_bridge/STATUS = {"Status":{"Module":0,"DeviceName":"Sonoff ZHA-Bridge","FriendlyName":["Sonoff ZHA-Bridge"],"Topic":"sonoff_zb_bridge","ButtonTopic":"0","Power":"0","PowerLock":"0","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
16:50:01.801 MQT: stat/sonoff_zb_bridge/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release-13.2.0/tasmota-zbbridge.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:33:02","StartupUTC":"2024-09-11T14:16:59","Sleep":50,"CfgHolder":4617,"BootCount":75,"BCResetTime":"2023-10-30T21:13:19","SaveCount":249,"SaveAddress":"F6000"}}
16:50:01.808 MQT: stat/sonoff_zb_bridge/STATUS2 = {"StatusFWR":{"Version":"14.2.0(release-tasmota)","BuildDateTime":"2024-08-14T12:36:35","Boot":31,"Core":"2_7_7","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"423/699"}}
16:50:01.815 MQT: stat/sonoff_zb_bridge/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Telia-1D882E",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00008080","00006000","00004000","00000000"]}}
16:50:01.829 MQT: stat/sonoff_zb_bridge/STATUS4 = {"StatusMEM":{"ProgramSize":648,"Free":352,"Heap":22,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"1540A1","FlashFrequency":40,"FlashMode":"DOUT","Features":["0809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080","00000000"],"Drivers":"1,2,!3,!4,!5,!6,7,!8,9,10,12,!16,!18,!19,!20,!21,!22,!24,26,!27,29,!30,!35,!37,!45,62,!68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
16:50:01.840 MQT: stat/sonoff_zb_bridge/STATUS5 = {"StatusNET":{"Hostname":"sonoff-zb-bridge-1263","IPAddress":"192.168.1.115","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"98:F4:AB:E3:24:EF","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
16:50:01.846 MQT: stat/sonoff_zb_bridge/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.8","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_E324EF","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
16:50:01.856 MQT: stat/sonoff_zb_bridge/STATUS7 = {"StatusTIM":{"UTC":"2024-09-11T14:50:01Z","Local":"2024-09-11T16:50:01","StartDST":"2024-03-31T01:00:00","EndDST":"2024-10-27T01:00:00","Timezone":99,"Sunrise":"06:36","Sunset":"19:48"}}
16:50:01.863 MQT: stat/sonoff_zb_bridge/STATUS10 = {"StatusSNS":{"Time":"2024-09-11T16:50:01"}}
16:50:01.870 MQT: stat/sonoff_zb_bridge/STATUS11 = {"StatusSTS":{"Time":"2024-09-11T16:50:01","Uptime":"0T00:33:02","UptimeSec":1982,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":25,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Telia-1D882E","BSSId":"A2:B5:3C:1D:68:2E","Channel":4,"Mode":"11n","RSSI":100,"Signal":-28,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Update Tasmota > 13.2.0 breakstheconnection  from Z2M Zigbee-Herdsman

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
I expect Z2M to be  able to connect properly to my Tasmotized Sonoff ZHA Bridge

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
The issue is that did not upgrade to the zbbridge binary, but instead upgraded away from it, no longer having Zigbee support. It should work if you make sure to use `tasmota-zbbridge.bin`

**Reply by oywino** [reply]
I do not understand what you mean, please explain?
The binary I use is `tasmota-zbbridge.bin.gz`  (v13.2.0 & v13.3.0 & v14.2.0)
But if by "zbbridge binary" you mean `ncp-uart-nsw zigbee radio chip firmware` then yes, I did upgrade to the latest version (ncp-uart-sw_6.7.8_115200).  It's been working just fine for a long time, but upgrading Tasmota made the zigbee-herdsman fail.

**Reply by sfromis** [reply, marked as answer]
No, that was not what I meant, but instead what I wrote. You need to upgrade Tasmota to
`http://ota.tasmota.com/tasmota/release/tasmota-zbbridge.bin.gz`
This is **not** what you did, as evidenced by seeing `"Version":"14.2.0(release-tasmota)"` in your `Status 0` output, where it should have been `release-zbbridge`.
The problem is that you instead upgraded to `tasmota.bin.gz`, not the correct one.

And this is not related to firmware of the Zigbee chip, if you have the version you mentioned, this should be just fine.

**Reply by oywino** [reply]
Aha - Now I understand what you mean. Of course, how stupid of me!!
You see, I used TasmoAdmin to  do the upgrade, and of course it's 'automatic' setting chooses the  default binary and not the zbbridge binary.  Thank you so very much for pointing this out 👍
Now I'll go ahead and choose the right binary, and hopefully everything will work now.
I'll post the result back here.

**Reply by sfromis** [reply]
BTW. When you only want to use Tasmota as a ser2net bridge to Z2M/ZHA, you really get no benefit from upgrading Tasmota to a recent version, as this is basically Tasmota doing nothing with the device.

**Reply by oywino** [reply]
I see, thanks for that info. 👍

**Comment by oywino** [comment]
Just confirming that after correcting to the right binary, as advised by @sfromis everything is now working as it should 👍
